types/wlr_cursor.c

```diff
@@ -733,3 +776,8 @@ void wlr_cursor_map_input_to_region(struct wlr_cursor *cur,
 
 	c_device->mapped_box = box;
 }
+
+void wlr_cursor_confine_to_region(struct wlr_cursor *cur,
+		struct wlr_box box) {
```

Comment:
you can't unset it, it's always active. You're supposed to use WLR_BOX_EMPTY if you do not want to confine it. I'll put this in the documentation. It doesn't take a pointer because that would just complicate things with allocations and such. A box is only 16 bytes anyway.

Reply:
-1. You shouldn't need to allocate anything even if you use pointers.

Reply:
How would I handle deallocation of the box here? Should wlr_cursor_confine_to_region deallocate the previous box? If so, what if something in rootston was using it? I still think it's simpler with a non-pointer.

Reply:
I've solved this problem by using a pointer to the region, since you can't have a pointer to a box inside the region.